### PR TITLE
Add EFI_TIMESTAMP_PROTOCOL

### PR DESCRIPTION
**Pull request by Ayush1325**
I am thinking of using it to implement `std::time::Instant`.

Signed-off-by: Ayush Singh <ayushsingh1325@gmail.com>

### CODE REVIEW

File: src/protocols/timestamp.rs [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    0x2e6e,
+    0x4262,
+    0xba,
+    0x65,
+    &[0x62, 0xb9, 0x23, 0x6e, 0x54, 0x95],
+);
+
+#[repr(C)]
+pub struct Properties {
+    frequence: u64,
```
Comment (dvdhrm) [inline src/protocols/timestamp.rs:17]: `frequence` -> `frequency`

File: src/protocols/timestamp.rs [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    0xafbfde41,
+    0x2e6e,
+    0x4262,
+    0xba,
+    0x65,
+    &[0x62, 0xb9, 0x23, 0x6e, 0x54, 0x95],
+);
+
+#[repr(C)]
+pub struct Properties {
```
Comment (dvdhrm) [inline src/protocols/timestamp.rs:16]: I think we should derive Clone, Copy, and Debug here.

File: src/protocols/timestamp.rs [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+pub type GetTimestamp = eficall! {fn() -> u64};
+pub type GetProperties = eficall! {fn(
+    *mut Properties
+) -> crate::base::Status};
+
+#[repr(C)]
+pub struct Protocol {
+    get_timestamp: GetTimestamp,
+    get_properties: GetProperties,
```
Comment (dvdhrm) [inline src/protocols/timestamp.rs:29]: These members should be marked `pub`, and I prefixed the types with `Protocol*` for namespacing.

**Review by dvdhrm — APPROVED** [review]
Minor comments inline. I fixed them up myself. If I screwed things up, or if you disagree, please let me know and we can discuss it in detail for the next release.

Thanks a lot!

**Comment by Ayush1325** [comment]
Thanks for fixing those mistakes.

**Pull request closed without merging**